core/chains/evm/gas/rollups/op_l1_oracle.go

```diff
@@ -276,14 +291,21 @@ func (o *optimismL1Oracle) checkForUpgrade(ctx context.Context) error {
 	if time.Since(o.upgradeCheckTs) < upgradePollingPeriod {
 		return nil
 	}
+
+	// Check in case somehow DA oracle address got reset to nil
+	if o.daOracleConfig.OracleAddress() == nil {
+		return errors.New("OracleAddress is required but was nil")
+	}
```

Comment:
Why don't we store the value in a field after checking in the constructor so that we don't have to repeat defensive checks.